Source/Core/DolphinQt/Config/Graphics/EnhancementsWidget.cpp

```diff
@@ -48,21 +49,28 @@ void EnhancementsWidget::CreateWidgets()
   auto* enhancements_layout = new QGridLayout();
   enhancements_box->setLayout(enhancements_layout);
 
-  m_ir_combo = new GraphicsChoice({tr("Auto (Multiple of 640x528)"), tr("Native (640x528)"),
-                                   tr("2x Native (1280x1056) for 720p"),
-                                   tr("3x Native (1920x1584) for 1080p"),
-                                   tr("4x Native (2560x2112) for 1440p"),
-                                   tr("5x Native (3200x2640)"), tr("6x Native (3840x3168) for 4K"),
-                                   tr("7x Native (4480x3696)"), tr("8x Native (5120x4224) for 5K")},
-                                  Config::GFX_EFB_SCALE);
-
-  if (g_Config.iEFBScale > 8)
+  // Only display the first 8 scales, which most users will not go beyond.
+  QStringList resolution_options{
+      tr("Auto (Multiple of 640x528)"),      tr("Native (640x528)"),
+      tr("2x Native (1280x1056) for 720p"),  tr("3x Native (1920x1584) for 1080p"),
+      tr("4x Native (2560x2112) for 1440p"), tr("5x Native (3200x2640)"),
+      tr("6x Native (3840x3168) for 4K"),    tr("7x Native (4480x3696)"),
+      tr("8x Native (5120x4224) for 5K")};
+  const int visible_resolution_option_count = static_cast<int>(resolution_options.size());
+
+  // If the current scale is greater than the max scale in the ini, add sufficient options so that
+  // when the settings are saved we don't lose the user-modified value from the ini.
+  const int max_efb_scale =
+      std::max(Config::Get(Config::GFX_EFB_SCALE), Config::Get(Config::GFX_MAX_EFB_SCALE));
+  for (int scale = static_cast<int>(resolution_options.size()); scale <= max_efb_scale; scale++)
   {
-    m_ir_combo->addItem(tr("Custom"));
-    m_ir_combo->setCurrentIndex(m_ir_combo->count() - 1);
+    resolution_options.append(QString::asprintf("%dx Native (%dx%d)", scale,
```

Comment:
```suggestion
resolution_options.append(QString::asprintf("%dx Native (%dx%d)", scale,
resolution_options.append(QString("%1x Native (%2x%3)").arg(QString::number(scale),
                                            QString::number(static_cast<int>(EFB_WIDTH) * scale),
                                            QString::number(static_cast<int>(EFB_HEIGHT) * scale));
```

Qt docs do not recommend using `asprintf` and recommend using `arg` instead.

Reply:
The string should use `tr`.